File: MATLAB_edition/frame2d_r_mat.m
function y = frame2d_r_mat( ele_info )
%FRAME2D_R_MAT Summary of this function goes here
%   Detailed explanation goes here
%   生成每个单元的转移矩阵,ele_info是[cos;sin]形式。
c=ele_info(1);
s=ele_info(2);
rr=[c,s,0;-s,c,0;0,0,1];
r=[rr,zeros(3,3);zeros(3,3),rr];
y=r;
end

File: Python_edition/frame2d_eg/frame2d_r_mat.py
import numpy as np
from math import *

def frame2d_r_mat(ele_info):

    #生成单元转换矩阵。
    c = ele_info[0]
    s = ele_info[1]
    rr = np.array([[c, s, 0],[-s, c, 0],[0, 0, 1]])
    r1=np.column_stack((rr,np.zeros((3, 3))))
    r2=np.column_stack((np.zeros((3, 3)),rr))
    r=np.row_stack((r1,r2))
    y = r

    return y